src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1521,13 +1785,22 @@ def enable_sequential_cpu_offload(self, gpu_id: Optional[int] = None, device: Un
             if not isinstance(model, torch.nn.Module):
                 continue
 
-            if name in self._exclude_from_cpu_offload:
-                model.to(device)
-            else:
-                # make sure to offload buffers if not all high level weights
-                # are of type nn.Module
-                offload_buffers = len(model._parameters) > 0
-                cpu_offload(model, device, offload_buffers=offload_buffers)
+            if component_device_mapping is None:
```

Comment:
Need to remove that, otherwise, we do nothing when component_device_mapping is None.
```suggestion
```